File: java/100. Same Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        if (p==null&&q==null) {
            return true;
        }else if (p==null||q==null) {
            return false;
        }
        if (p.val!=q.val) {
            return false;
        }else{
            return isSameTree(p.left, q.left)&&isSameTree(p.right, q.right);
        }
        
    }
}

File: ruby/100. Same Tree.rb
# Definition for a binary tree node.
# class TreeNode
#     attr_accessor :val, :left, :right
#     def initialize(val)
#         @val = val
#         @left, @right = nil, nil
#     end
# end

# @param {TreeNode} p
# @param {TreeNode} q
# @return {Boolean}
def is_same_tree(p, q)
    return (p==nil && q==nil) || (p!=nil && q!=nil && p.val==q.val && is_same_tree(p.left, q.left) && is_same_tree(p.right, q.right))
end